xonsh/cli_utils.py

```diff
@@ -371,6 +373,149 @@ def dispatch(parser: ap.ArgumentParser, args=None, lenient=False, **ns):
     return _dispatch_func(func, ns)
 
 
+class ArgparseCompleter:
+    """A completer function for ArgParserAlias commands"""
+
+    def __init__(self, parser: ap.ArgumentParser, command, **kwargs):
+        args = tuple(c.value for c in command.args[: command.arg_index])
+
+        self.parser, self.remaining_args = self.get_parser(parser, args[1:])
+
+        self.command = command
+        kwargs["command"] = command
+        self.kwargs = kwargs
+        """will be sent to completer function"""
```

Comment:
should this line be here or did it get cut from somewhere else?

Reply:
the comment is correct. The kwargs will be used to call the completer functions

Reply:
Ok, can you make it a comment and not a string then, please?

Reply:
done.